android/app/src/main/java/com/example/makeat/ui/navigation/NavItems.kt

```diff
@@ -0,0 +1,16 @@
+package com.example.makeat.ui.navigation
+
+import com.example.makeat.R
+import com.example.makeat.utils.Drawable
+
+data class NavItem(
+    val label: String, val route: Routes, val icon: Drawable
```

Comment:
Là faire attention à Drawable que t'as défini dans utils/drawable qui a le même nom que [Drawable](https://developer.android.com/reference/android/graphics/drawable/Drawable) qui m'a fait buger car je trouvait pas toImageVector() lol et elle accepte une image bitmap comme une vectorielle (svg etc) et ça serait risqué (pas possible normalement) d'appeller toImageVector() sur une ressource qui pourrait être une bitmap et tant ça bug et tout derrière

Tu peut faire en sorte que la navbar accepte un vrai Drawable et donc image normal ou svg mais je pense que tu peux utiliser ImageVector.vectorResource() là où tu définis les routes et juste faire icon:item.icon dans la navbar
Enfin je sais pas ce que t'as voulu faire lol

Reply:
Initialement, c'est ce que j'avais voulu faire. Mais `ImageVector.vectorResource()` est un `@Composable` et celui qui utilise une fonction de composition dois également être une fonction de composition. Donc j'avais fait comme tu dis, mais je me suis retrouvé avec quelque chose dans ce goût-là:  
```suggestion
    package com.example.makeat.ui.navigation

import androidx.compose.runtime.Composable
import com.example.makeat.R
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.res.vectorResource


data class NavItem(
    val label: String, val route: Routes, val icon: ImageVector
)

@Composable
fun bottomNavItems(): List<NavItem> {
    return listOf(
        NavItem("Recipes", Routes.Recipes, ImageVector.vectorResource(R.drawable.chef_hat)),
        NavItem("Plan", Routes.Plan, ImageVector.vectorResource(R.drawable.calendar)),
        NavItem("Stock", Routes.Stock, ImageVector.vectorResource(R.drawable.box)),
        NavItem("Profile", Routes.Profile, ImageVector.vectorResource(R.drawable.user)),
    )}
}
```

Et ça me paraissait étrange de définir une fonction de composition qui n'est pas un élément d'UI.